File: modules/common/src/main/java/com/media/bus/common/utils/MessageUtil.java
package com.media.bus.common.utils;

import lombok.extern.slf4j.Slf4j;
import org.springframework.context.NoSuchMessageException;
import org.springframework.context.support.ResourceBundleMessageSource;

import java.util.Locale;

/**
 * Message Bundle 관련 Utility
 */
@Slf4j
public class MessageUtil {
	
	private static final ResourceBundleMessageSource messageSource = new ResourceBundleMessageSource();
	static {
		messageSource.setBasenames("messages/message", "messages/common-message");
		messageSource.setDefaultEncoding("UTF-8");
	}
	
	private MessageUtil() {
		throw new IllegalStateException("Utility class");
	}
	
	public static String getMessage(String messageId) {
		return getMessage(messageId, null);
	}
	
	public static String getMessage(String messageId, Object[] objects) {
		return getMessage(messageId, objects, Locale.getDefault());
	}
	
	public static String getMessage(String messageId, Object[] objects, Locale locale) {
		try {
			return messageSource.getMessage(messageId, objects, locale);
		} catch(NoSuchMessageException e) {
			log.error("No message found under code '{}' for locale '{}'.", messageId, locale);
			return "";
		}
	}

}

File: modules/common/src/main/kotlin/com/media/bus/common/utils/MessageUtil.kt
package com.media.bus.common.utils

import org.slf4j.LoggerFactory
import org.springframework.context.NoSuchMessageException
import org.springframework.context.support.ResourceBundleMessageSource
import java.util.*

/**
 * ## Message Bundle 관련 Utility
 */
object MessageUtil {

    private val log = LoggerFactory.getLogger(MessageUtil::class.java)

    private val messageSource = ResourceBundleMessageSource().apply {
        setBasenames("messages/message", "messages/common-message")
        setDefaultEncoding("UTF-8")
    }

    @JvmStatic
    fun getMessage(messageId: String): String = getMessage(messageId, null)

    @JvmStatic
    fun getMessage(messageId: String, objects: Array<Any>?): String =
        getMessage(messageId, objects, Locale.getDefault())

    @JvmStatic
    fun getMessage(messageId: String, objects: Array<Any>?, locale: Locale): String =
        try {
            messageSource.getMessage(messageId, objects, locale)
        } catch (e: NoSuchMessageException) {
            log.error("No message found under code '{}' for locale '{}'.", messageId, locale)
            ""
        }
}
